File: com-python-math/src/main/java/com/titanic/math/martix/VectorMath.py
import numpy as np

u = np.array([[1, 2, 3]]).T
v = np.array([[5, 6, 7]]).T

'向量相加'
print(u + v)

'向量*数'
print(3 * u.T)

uv = np.array([3, 5, 2])
vv = np.array([1, 4, 7])

'向量內积'
print(np.dot(uv, vv))

uc = np.array([3, 5])
vc = np.array([1, 4])

'二维外积'
print(np.cross(uc,vc))


ux = np.array([[1,2,3]]).T
vx = np.array([[4,5,6]]).T
wx = np.array([[7,8,9]]).T

'向量的线性组合'
print(3*ux+4*vx+5*wx)

File: com-java-math/src/main/java/com/titanic/math/matrix/VectorMath.java
package com.titanic.math.matrix;

import org.nd4j.linalg.api.ndarray.INDArray;
import org.nd4j.linalg.factory.Nd4j;

public class VectorMath
{
    public static void main(String[] args)
    {
        vectorAdd();
    }

    public static void addi()
    {
        INDArray u = Nd4j.create(new double[]{1, 2, 3});
        INDArray v = Nd4j.create(new double[]{5, 6, 7});

        v.addi(u);

        System.out.println(v);

        u.muli(3);
        System.out.println(u);
    }

    /**
     * 向量內积
     */
    public static void dot()
    {
        INDArray u = Nd4j.create(new double[]{3, 5, 2});
        INDArray v = Nd4j.create(new double[]{1, 4, 7});

        u.muli(v);
        System.out.println(u.sumNumber().doubleValue());

    }

    /**
     * 向量外积
     */
    public static void cross()
    {
        INDArray u = Nd4j.create(new double[]{3, 5});
        INDArray v = Nd4j.create(new double[]{1, 4});


    }

    /**
     * 向量的线性组合
     */
    public static void vectorAdd()
    {
        INDArray u = Nd4j.create(new double[]{1, 2, 3});
        INDArray v = Nd4j.create(new double[]{4, 5, 6});
        INDArray w = Nd4j.create(new double[]{7, 8, 9});

        INDArray x = u.muli(3).add(v.muli(4)).add(w.muli(5)).dup();
        System.out.println(x);
    }
}
